// react-router-auth/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = _interopDefault(require('react'));
var PropTypes = _interopDefault(require('prop-types'));
var reactRouterDom = require('react-router-dom');

var _extends = Object.assign || function (target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i];

    for (var key in source) {
      if (Object.prototype.hasOwnProperty.call(source, key)) {
        target[key] = source[key];
      }
    }
  }

  return target;
};













var objectWithoutProperties = function (obj, keys) {
  var target = {};

  for (var i in obj) {
    if (keys.indexOf(i) >= 0) continue;
    if (!Object.prototype.hasOwnProperty.call(obj, i)) continue;
    target[i] = obj[i];
  }

  return target;
};

var AuthRoute = function AuthRoute(_ref) {
  var Component = _ref.component,
      authenticated = _ref.authenticated,
      redirectTo = _ref.redirectTo,
      rest = objectWithoutProperties(_ref, ['component', 'authenticated', 'redirectTo']);

  return React.createElement(reactRouterDom.Route, _extends({}, rest, {
    render: function render(props) {
      return authenticated ? React.createElement(Component, props) : React.createElement(reactRouterDom.Redirect, { to: {
          pathname: redirectTo,
          state: { from: props.location }
        }
      });
    }
  }));
};

AuthRoute.propTypes = {
  authenticated: PropTypes.bool,
  redirectTo: PropTypes.string.isRequired,
  component: PropTypes.func.isRequired
};

AuthRoute.defaultProps = {
  authenticated: false
};

var UnauthRoute = function UnauthRoute(_ref) {
  var Component = _ref.component,
      authenticated = _ref.authenticated,
      redirectTo = _ref.redirectTo,
      rest = objectWithoutProperties(_ref, ['component', 'authenticated', 'redirectTo']);

  return React.createElement(reactRouterDom.Route, _extends({}, rest, {
    render: function render(props) {
      return authenticated ? React.createElement(reactRouterDom.Redirect, { to: {
          pathname: redirectTo,
          state: { from: props.location }
        }
      }) : React.createElement(Component, props);
    }
  }));
};

UnauthRoute.propTypes = {
  authenticated: PropTypes.bool,
  redirectTo: PropTypes.string.isRequired,
  component: PropTypes.func.isRequired
};

UnauthRoute.defaultProps = {
  authenticated: false
};

exports.AuthRoute = AuthRoute;
exports.UnauthRoute = UnauthRoute;
